src/extension.ts

```diff
@@ -357,63 +370,51 @@ export function activate(context: ExtensionContext): void {
     }
 
     const uri = document.uri;
-    // Untitled files go to a default client.
-    if (uri.scheme === "untitled" && !defaultClient) {
-      // Create the language client and start the client.
-      defaultClient = new LanguageClient(
-        "elixirLS", // langId
-        "ElixirLS", // display name
-        serverOptions,
-        clientOptions
-      );
-      const disposable = defaultClient.start();
-
-      // Push the disposable to the context's subscriptions so that the
-      // client can be deactivated on extension deactivation
-      context.subscriptions.push(disposable);
-      return;
-    }
-
     let folder = workspace.getWorkspaceFolder(uri);
-    // Files outside a folder can't be handled. This might depend on the language.
-    // Single file languages like JSON might handle files outside the workspace folders.
+
+    // Files outside of workspace go to default client when no directory is open
+    // otherwise they go to first workspace
+    // (even if we pass undefined in clientOptions vs will pass first workspace as rootUri/rootPath)
     if (!folder) {
-      return;
+      if (workspace.workspaceFolders && workspace.workspaceFolders.length !== 0) {
+        // untitled file assigned to first workspace
+        folder = workspace.getWorkspaceFolder(workspace.workspaceFolders[0].uri)!;
+      } else {
+        // no workspace folders - use default client
+        if (!defaultClient) {
+          // Create the language client and start the client.
+          defaultClient = startClient(context, clientOptions);
+        }
+        return;
+      }
     }
-
+    
     // If we have nested workspace folders we only start a server on the outer most workspace folder.
     folder = getOuterMostWorkspaceFolder(folder);
-
+    
     if (!clients.has(folder.uri.toString())) {
+      const pattern = `${folder.uri.fsPath}/**/*`
+      // untitled files to first workspace
+      const unttled = folder.index === 0 ? [
+        { language: "elixir", scheme: "untitled" },
+        { language: "eex", scheme: "untitled" },
+        { language: "html-eex", scheme: "untitled"}
+      ] : [];
       const workspaceClientOptions: LanguageClientOptions = Object.assign(
         {},
         clientOptions,
         {
           documentSelector: [
```

Comment:
```suggestion
          // the client will iterate through this list and chose the first matching element
          documentSelector: [
```
I think a comment like the above would be helpful in understanding how this code works

Reply:
+1